src/effects/smoke_burst.c

```diff
@@ -8,7 +8,17 @@ INCLUDE_ASM(s32, "effects/smoke_burst", smoke_burst_main);
 void smoke_burst_init(void) {
 }
 
-INCLUDE_ASM(s32, "effects/smoke_burst", smoke_burst_update);
+void smoke_burst_update(EffectInstance *effect) {
+    SmokeBurstFXData* fxData = effect->data.smokeBurst;
```

Comment:
this temp is typically called just `data`. And the casts below shouldn't be needed